service/storage_service.cc

```diff
@@ -1841,6 +1841,40 @@ future<> storage_service::do_stop_stream_manager() {
     });
 }
 
+future<> storage_service::node_ops_cmd_heartbeat_updater(std::string ops, utils::UUID uuid, std::list<gms::inet_address> nodes, lw_shared_ptr<bool> heartbeat_updater_done) {
+    return seastar::async([this, ops, uuid, nodes = std::move(nodes), heartbeat_updater_done] {
+        slogger.info("{}[{}]: Started heartbeat_updater", ops, uuid);
+        node_ops_cmd cmd;
+        if (ops == "decommission") {
+            cmd = node_ops_cmd::decommission_heartbeat;
```

Comment:
How about making cmd a parameter instead of this switch?

Reply:
I wanted to avoid bring node_ops_cmd (defined in repair.hh) into service/storage_service.hh which has the node_ops_cmd_heartbeat_updater signature. 

We need the switch to figure out the ops name for the log even if we pass the node_ops_cmd to storage_service::node_ops_cmd_heartbeat_updater.

Reply:
It's not the right place to stick it. You can create a new header for node_ops_cmd, and define operator<<(std::ostream&, node_ops_cmd). Then you don't have to bring in repair.hh and we don't have to repeat the switch every place.

Reply:
The problem with operator<<(std::ostream&, node_ops_cmd) is:

We need to print something like:

storage_service - **replace**[db9f0ae6-1018-4d8a-b048-9ff4a6960da1] 

But the print operator for node_ops_cmd will give 

storage_service - **node_ops_cmd::replace_heartbeat**[db9f0ae6-1018-4d8a-b048-9ff4a6960da1] 

or

storage_service - **replace_heartbeat**[db9f0ae6-1018-4d8a-b048-9ff4a6960da1]

Reply:
ok. But at least don't pass the op as a string.

Reply:
Ok. Fixed.